core/server/apps/amp/lib/router.js

```diff
@@ -42,21 +43,42 @@ function getPostData(req, res, next) {
             next();
         })
         .catch(function (err) {
-            if (err instanceof errors.NotFoundError) {
-                return next(err);
-            }
-
             next(err);
         });
 }
 
+function checkIfAMPIsEnabled(req, res, next) {
+    var ampIsEnabled = config.theme.amp;
+
+    if (ampIsEnabled) {
+        return next();
+    }
+
+    // CASE: we don't support amp pages for static pages
+    if (req.body.post && req.body.post.page) {
+        return errors.error404(req, res, next);
+    }
+
+    /**
+     * CASE: amp is disabled, we redirect the user to the requested post
```

Comment:
Why are we redirecting if AMP is disabled? It should be a 404 in my opinion, a 301/302 is saying "yes we have an AMP page but it's located here". It's similar to the soft-404 problems we've had in the past.